includes/class-coblocks-labs.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Register CoBlocks Labs
+ *
+ * @package CoBlocks
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Main @@pkg.title Class
+ *
+ * @since 2.0.0
+ */
+class CoBlocks_Labs {
+	/**
+	 * This plugin's instance.
+	 *
+	 * @var CoBlocks_Labs
+	 */
+	private static $instance;
+
+	/**
+	 * Registers the plugin.
+	 *
+	 * @return CoBlocks_Labs
+	 */
+	public static function register() {
+		if ( null === self::$instance ) {
+			self::$instance = new CoBlocks_Labs();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+
+		add_action( 'init', array( $this, 'register_settings' ) );
+		add_action( 'init', array( $this, 'propagate_settings' ) );
+		add_action( 'enqueue_block_editor_assets', array( $this, 'editor_assets' ) );
+
+	}
+
+	/**
+	 * Go theme is active.
+	 *
+	 * @since 2.24.4
+	 * @return boolean
+	 */
+	public static function is_go_theme_active() {
+		$template_part      = explode( 'themes/', esc_attr( wp_get_theme( get_template() )->get_template() ) );
+		$active_template    = isset( $template_part ) && isset( $template_part[0] ) ? $template_part[0] : false;
+		$active_theme_is_go = 'go' === $active_template;
+		return $active_theme_is_go;
+	}
+
+	/**
+	 * Go theme is installed.
+	 *
+	 * @since 2.22.4
+	 * @return string
+	 */
+	public static function is_go_theme_installed() {
+		$go_theme_path   = get_theme_root() . '/go/index.php';
+		$is_go_installed = file_exists( $go_theme_path );
+		return $is_go_installed;
+	}
+
+	/**
+	 * Site themes.php page.
+	 *
+	 * @since 2.22.4
+	 * @return string
+	 */
+	public static function get_themes_php_uri() {
+		return admin_url( 'themes.php' );
+	}
```

Comment:
I don't think we need this helper function since it's only being used in one place. I think we can remove a lot of these helpers used inside of `get_coblocks_labs_data()` and just return in the data we need without the helpers:

Also, to check if the launch guide should be enabled, maybe we can just check if the `wpnux_export_data` is empty of not, similar to how we do in nextgen.

```php
! empty( get_option( 'wpnux_export_data' ) );
```

```php
/**
 * Function get_coblocks_labs_data is used to set a global object
 * `coblocksLabs` with relevant data.
 *
 * @since 2.22.4
 * @return array Array of relevant CoBlocks labs data.
 */
public static function get_coblocks_labs_data() {
	return array(
		'isGoThemeActive'      => defined( 'GO_VERSION' ) && 'go' === get_option( 'stylesheet' ),
		'isGoThemeInstalled'  => defined( 'GO_VERSION' ) && 'go' === get_option( 'stylesheet' ),
		'goThemeInstallUri'     => admin_url( 'theme-install.php?theme=go' ),
		'goThemeDetailsUri'   => admin_url( 'themes.php' ) . '?theme=go',
		'launchGuideEligible'  => ! empty( get_option( 'wpnux_export_data' ) ),
		'isLabsEnabled'           => apply_filters( 'coblocks_labs_controls_enabled', false ),
	);
}
```
Or defined them in variables above the return.

```php
/**
 * Function get_coblocks_labs_data is used to set a global object
 * `coblocksLabs` with relevant data.
 *
 * @since 2.22.4
 * @return array Array of relevant CoBlocks labs data.
 */
public static function get_coblocks_labs_data() {
	$is_go_active                       = defined( 'GO_VERSION' ) && 'go' === get_option( 'stylesheet' );
        $is_launch_guide_enabled = get_option( 'wpnux_export_data' ) === false ? false : true;
	
         return array(
		'isGoThemeActive'      => $is_go_active,
		'isGoThemeInstalled'  => $is_go_active,
		'goThemeInstallUri'     => admin_url( 'theme-install.php?theme=go' ),
		'goThemeDetailsUri'   => admin_url( 'themes.php' ) . '?theme=go',
		'launchGuideEligible'  => ! empty( get_option( 'wpnux_export_data' ) ),
		'isLabsEnabled'           => apply_filters( 'coblocks_labs_controls_enabled', false ),
	);
}
```

Do we need to both check `isGoThemeActive` and `isGoThemeInstalled` or can we just check if it's active?

I think we can clean up some of these helpers since they're only used one time in this class and aren't going to be used outside of them. That will just help trim down the length of this file.

Reply:
I forgot that the go theme installation status is an AC because we need to include a conditional URI to either install or activate depending on the install status.